### PR TITLE
Initial `tracing` crate support

### PR DESCRIPTION
**Pull request by GabrielMajeri**
**Connections**

Initial work towards #491 "CPU profiling infrastructure"

**Description**

Adds a new optional feature to `wgpu-core` called `profiling`, which pulls in the `tracing` crate as a dependency.

As a test, I've added a profiling span to the `queue_submit` function.

**Note:** I'm using the `WARN` level for the events, because otherwise all of the logging/debug messages would also appear in my terminal. I'm not sure how to filter only for `tracing` messages.

**Testing**

To test locally, I've modified the `framework.rs` file from `wgpu-rs/examples`. It now looks like:
```rust
tracing_subscriber::fmt()
   .with_env_filter("[queue_submit]=warn")
   .init();

//env_logger::init();
futures::executor::block_on(run_async::<E>(event_loop, window));
```

Running the `boids` examples gives me lines such as
```
Jun 18 17:38:59.021  WARN queue_submit: wgpu_core::device::queue: submitting to queue
Jun 18 17:38:59.022  WARN queue_submit: wgpu_core::device::queue: finished submission
```
from where I can deduce it takes 1ms to perform a queue submission.

Feedbacks and ideas are welcome!

### CODE REVIEW

**Comment by cwfitzgerald** [comment]
This looks like a good starting place! I was actually already working on tracing support (we can still merge this in though). Would you mind hopping on the matrix (https://matrix.to/#/#wgpu:matrix.org) so we can discuss distribution of work etc.

File: wgpu-core/Cargo.toml [hunk 1]
```diff
@@ -16,10 +16,11 @@ license = "MPL-2.0"
 
 [features]
 default = []
-trace = ["ron", "serde", "wgt/trace"]
-replay = ["serde", "wgt/replay"]
 #NOTE: glutin feature is not stable, use at your own risk
 #glutin = ["gfx-backend-gl/glutin"]
+profiling = ["tracing"]
```
Comment (kvark) [inline wgpu-core/Cargo.toml:21]: We talked about this on the Matrix, and I think we should just unconditionally switch from `log` to `tracing`. Not making it optional. The *subscribers* will be optional wherever we hook them up.

File: wgpu-core/Cargo.toml [hunk 2]
```diff
@@ -37,6 +38,7 @@ ron = { version = "0.5", optional = true }
 serde = { version = "1.0", features = ["serde_derive"], optional = true }
 smallvec = "1"
 spirv_headers = { version = "1.4.2" }
+tracing = { version = "0.1.15", optional = true }
```
Comment (kvark) [inline wgpu-core/Cargo.toml:41]: do we need "0.1.15" specifically? or just "0.1"?

File: wgpu-core/src/device/queue.rs [hunk 3]
```diff
@@ -365,6 +367,14 @@ impl<G: GlobalIdentityHandlerFactory> Global<G> {
         queue_id: id::QueueId,
         command_buffer_ids: &[id::CommandBufferId],
     ) {
+        #[cfg(feature = "tracing")]
+        let span = span!(Level::TRACE, "queue_submit");
```
Comment (kvark) [inline wgpu-core/src/device/queue.rs:371]: can we just put the `[#instrument]` attribute on the function itself?

**Comment by GabrielMajeri** [comment]
@cwfitzgerald I'm closing this, feel free to reuse the code

**Pull request closed without merging**